### PR TITLE
Add codecov to CI and to readme

### PR DESCRIPTION
**Pull request by BenediktBurger**
Adds codecov to the CI, such that test coverage is calculated and uploaded.
PRs will get a comment regarding the code coverage

The readme shows a test coverage badge.

### CODE REVIEW

**Comment by BenediktBurger** [comment]
@cjermain, could you add the codecov token to the actions secrets, please?

**Comment by BenediktBurger** [comment]
Finally I found the culprit in order to create coverage reports: you have to install pymeasure as an editable install.

As I tried a lot, this PR should be squashed into one commit.

File: .github/workflows/pymeasure_CI.yml [hunk 1]
```diff
@@ -46,6 +46,47 @@ jobs:
...
+        uses: mamba-org/setup-micromamba@v1
+        with:
+          environment-file: .github/pymeasure.yml
+          create-args: python=${{ matrix.python-version }}
+          cache-environment-key: py${{ matrix.python-version }}-${{ matrix.os }}-mamba-${{ env.CACHE_NUMBER }}-${{ hashFiles('environment.yml') }}
+          cache-downloads: false
+      - name: Python version
+        run: python --version
+      - name: Install Pymeasure
+        # If the pytest problem matcher stops working because of bad paths, do an editable install
```
Comment (CasperSchippers) [inline .github/workflows/pymeasure_CI.yml:70]: This line seems like a copy left-over?

File: .github/workflows/pymeasure_CI.yml [hunk 2]
```diff
@@ -46,6 +46,47 @@ jobs:
...
+        run: pip install -e .[tests]  # editable for covtest
+      - name: Pymeasure version
+        run: python -c "import pymeasure;print(pymeasure.__version__)"
+      - name: Test for Coverage
+        # xvfb for graphical interface
+        run: |
+          echo "::add-matcher::.github/pytest.json"
+          xvfb-run -a pytest --cov=pymeasure --cov-report=xml
+      - name: Upload coverage to Codecov
```
Comment (CasperSchippers) [inline .github/workflows/pymeasure_CI.yml:79]: One thing I wonder: for the coverage badge, can codecov distinguish between a push to master and a PR, or does it show the latest uploaded coverage report (regardless of whether it is from a PR)?

**Review by CasperSchippers — APPROVED** [review]
I think this is good idea! Hopefully it will increase the coverage, and at least knowing the coverage won't hurt. Thanks for adding this.

File: .github/workflows/pymeasure_CI.yml (hunk 1 above)
Comment (BenediktBurger) [reply .github/workflows/pymeasure_CI.yml]: Done

**Pull request merged**